src/Configuration/OnlyRuleResolver.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Configuration;
+
+use Rector\Contract\Rector\RectorInterface;
+use Rector\Exception\Configuration\RectorRuleNotFoundException;
+
+/**
+ * @see \Rector\Tests\Configuration\OnlyRuleResolverTest
+ */
+final class OnlyRuleResolver
+{
+    /**
+     * @param RectorInterface[] $rectors
+     */
+    public function __construct(
+        private readonly array $rectors
+    ) {
+    }
+
+    public function resolve(string $rule): string
+    {
+        //fix wrongly double escaped backslashes
+        $rule = str_replace('\\\\', '\\', $rule);
+
+        //remove single quotes appearing when single-quoting arguments on windows
+        if (str_starts_with($rule, "'") && str_ends_with($rule, "'")) {
+            $rule = substr($rule, 1, -1);
+        }
+
+        $rule = ltrim($rule, '\\');
+
+        foreach ($this->rectors as $rector) {
+            if ($rector::class === $rule) {
+                return $rule;
+            }
+        }
```

Comment:
@TomasVotruba how about add only suffix rule for last resort search, if no `\`, just use as last resort for any rule with `\SomeRector.php` suffix:

```php
        foreach ($this->rectors as $rector) {
            if (str_ends_with($rector::class, "\\" . $rule)) {
                return $rule;
            }
        }
```

most of the time, in favor of fast copy paste fast patch, copy paste rule name from the list seems easier, while it have drawback if there is conflict name, it only got the first one match